app/src/main/java/io/punchtime/punchtime/ui/fragments/SettingsFragment.java

```diff
@@ -115,13 +119,26 @@ public boolean onPreferenceClick(Preference preference) {
         // TODO: 16/05/16 show contact of company
     }
 
-    protected static void setCompaniesData(ListPreference lp) {
-        // TODO: 16/05/16 get companies from firebase
-        CharSequence[] entries = { "English", "French" };
-        CharSequence[] entryValues = {"1" , "2"};
-        lp.setEntries(entries);
-        lp.setDefaultValue("1");
-        lp.setEntryValues(entryValues);
+    protected void setCompaniesData(final ListPreference lp) {
+        activity.getFirebaseRef().child("users").child(activity.getAuth().getUid()).child("employer").addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                List<CharSequence> entries = new ArrayList<>();
+                List<CharSequence> entryValues = new ArrayList<>();
+                Iterable<DataSnapshot> children = dataSnapshot.getChildren();
+                for (DataSnapshot i : children) {
+                    entries.add(i.toString());
+                    entryValues.add(i.getKey());
```

Comment:
I just realised that it's not the key I need to get, but that the `i.toString()` *is* the key and that the value needs to be gotten from the employers part in firebase. Maybe we should copy the name inside here too, that's easier (also for the Dashboard I use that, but firebase operations are practically free in javascript).